pkg/gomega/assertions/eventually_csv.go

```diff
@@ -0,0 +1,41 @@
+package assertions
+
+import (
+	"context"
+	"log"
+
+	"github.com/onsi/gomega"
+	operatorv1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+)
+
+// EventuallyCsv returns true if given namespace contains given CSV spec display name, and the CSV status is "Succeeded"
+// It can be used with the standard or custom gomega matchers, timeout and polling interval
+//
+//	EventuallyCsv(ctx, dynamicClient, operatorName, namespaceName).Should(BeTrue())
+func EventuallyCsv(ctx context.Context, dynamicClient dynamic.DynamicClient, specDisplayName, namespace string) gomega.AsyncAssertion {
```

Comment:
```suggestion
// EventuallyCSV returns true if given namespace contains given CSV spec display name, and the CSV status is "Succeeded"
// It can be used with the standard or custom gomega matchers, timeout and polling interval
//
//	EventuallyCSV(ctx, dynamicClient, operatorName, namespaceName).Should(BeTrue())
func EventuallyCSV(ctx context.Context, dynamicClient dynamic.DynamicClient, specDisplayName, namespace string) gomega.AsyncAssertion {
```

Reply:
Also, interested in understanding why this should be an async assertion rather than just an assertion. Happy to talk through it @ritmun

Reply:
Yeah! I think this is worth a larger discussion as I've been having some thoughts about the install tests recently but when writing these helpers we have two different options: `AsyncAssertion` and an `Assertion`. This could also be something like below keeping the same function but requiring this be something immediately fulfilled.

```go
func ExpectCSV(...) gomega.Assertion {
```

I only bring this up because I'm starting to have concerns over the value of some of the checks I've been writing (like `EventuallyDeployment`) which _should_ never fail if the healthcheck passes. I haven't proved this out yet, just having a hunch.

I am fine with us writing some form of this check, just want to understand why we pick one over the other.